### PR TITLE
[Fix bug 1254127] Make footer sticky to the bottom

### PR DESCRIPTION
**Pull request by comzeradd**

### CODE REVIEW

**Comment by comzeradd** [comment]
@akatsoulas r?

**Comment by johngian** [comment]
Please rebase with master.

File: mozillians/static/mozillians/css/main.less [hunk 1]
```diff
@@ -133,8 +140,38 @@
 .html-rtl .right{float:left;}
 .clear{clear:both;}
 
+html, body {
+  height: 100%;
```
Comment (johngian) [inline mozillians/static/mozillians/css/main.less]: I don't see this getting overridden somehow. Isn't this the default value?

File: mozillians/static/mozillians/css/main.less [hunk 2]
```diff
@@ -133,8 +140,38 @@
...
+}
+
 body {
     #content-wrapper { background: @white; }
+    #outer-wrapper {
+        height: 100%;
```
Comment (johngian) [inline mozillians/static/mozillians/css/main.less]: Same here.

File: mozillians/static/mozillians/css/main.less [hunk 3]
```diff
@@ -133,8 +140,38 @@
...
+    }
+    #page {
+        min-height: 100%;
```
Comment (johngian) [inline mozillians/static/mozillians/css/main.less]: Same here.

File: mozillians/static/mozillians/css/main.less [hunk 4]
```diff
@@ -133,8 +140,38 @@
...
+        margin-bottom: -@footerDesktop;
+        @media (max-width: @breakTablet) {
+            margin-bottom: -@footerTablet;
+        }
+        @media (max-width: @breakMobileLandscape) {
+            margin-bottom: -@footerMobile;
+        }
+    }
+    #page:after {
+        content: "";
```
Comment (johngian) [inline mozillians/static/mozillians/css/main.less]: How does this affect the sticky footer?

File: mozillians/static/mozillians/css/main.less [hunk 5]
```diff
@@ -133,8 +140,38 @@
...
+        display: block;
+        height: @footerDesktop;
+        background-color: @white;
```
Comment (johngian) [inline mozillians/static/mozillians/css/main.less]: Same here.

File: mozillians/static/mozillians/css/main.less [hunk 6]
```diff
@@ -133,8 +140,38 @@
...
+        min-height: 100%;
+        margin-bottom: -@footerDesktop;
+        @media (max-width: @breakTablet) {
+            margin-bottom: -@footerTablet;
+        }
+        @media (max-width: @breakMobileLandscape) {
+            margin-bottom: -@footerMobile;
+        }
+    }
+    #page:after {
```
Comment (johngian) [inline mozillians/static/mozillians/css/main.less]: Can't we just add a placeholder element in HTML instead of adding ``:after`` content?

Comment (johngian) [inline mozillians/static/mozillians/css/main.less]: I am not sure what's better (element vs pseudoelement) please add a comment in the code to explain why we need this.

File: mozillians/static/mozillians/css/main.less (hunk 1 above)
Comment (comzeradd) [inline mozillians/static/mozillians/css/main.less]: Not on all browsers.

File: mozillians/static/mozillians/css/main.less (hunk 2 above)
Comment (comzeradd) [inline mozillians/static/mozillians/css/main.less]: On div elements the default height is usually auto (depends on the browser), not 100%.

File: mozillians/static/mozillians/css/main.less (hunk 6 above)
Comment (comzeradd) [inline mozillians/static/mozillians/css/main.less]: Yes, in general there are two ways of doing this. Either using an empty element with specific height or use an `:after` selector to the existing page wrapper. Both have the same effect. I choose the latter because our wrappers structure permits it and it allows us to keep the html code cleaner.

Comment (johngian) [inline mozillians/static/mozillians/css/main.less]: Sounds good. Let's add a comment here to explain this for future reference.

File: mozillians/static/mozillians/css/main.less [hunk 7]
```diff
@@ -133,8 +140,38 @@
 .html-rtl .right{float:left;}
 .clear{clear:both;}
 
+html, body {
+  height: 100%;
+}
+
 body {
     #content-wrapper { background: @white; }
+    #outer-wrapper {
```
Comment (johngian) [inline mozillians/static/mozillians/css/main.less]: Let's concat this one with the same definition above.

**Comment by comzeradd** [comment]
PR rebased to master

File: mozillians/static/mozillians/css/main.less [hunk 8]
```diff
@@ -133,8 +140,38 @@
...
+        @media (max-width: @breakMobileLandscape) {
+            margin-bottom: -@footerMobile;
+        }
+    }
+    // Placeholder for pushing the footer
+    #page:after {
+        content: "";
+        display: block;
+        height: @footerDesktop;
+        background-color: @white;
```
Comment (johngian) [inline mozillians/static/mozillians/css/main.less]: let's switch the order so ``height`` is right next to the media queries that also change ``height``.

**Comment by akatsoulas** [comment]
The profile view has a line right above the footer.
http://awesomescreenshot.com/0da5rm3lea

**Comment by akatsoulas** [comment]
There is a line in profile_view right above the footer
profile_view

**Comment by akatsoulas** [comment]
The main page when visiting the site seems broken

main_page_logged_out

**Comment by comzeradd** [comment]
@akatsoulas  I fixed the first issue. The problem was that we have the sandstone background to the whole website, instead of just the header where we actually use it.

The second problem is caused due to the fact that the "Get Involved" section is not sticky on its own. In order to stick it to the bottom we should move outside the `#page` wrapper and chage heights and margins specifically for this case (home page and not logged in).

That significantly increases the complexity of the code, but I agree it looks better. I added a second commit for that to see what I mean. I'll squash it if we decide to go this way.

File: mozillians/jinja2/base.html [hunk 9]
```diff
@@ -179,6 +179,8 @@
     {% endblock %}
   </div><!-- close #wrapper -->
 
+  {% block get_involved %}{% endblock %}
+
     {% block site_footer %}
```
Comment (johngian) [inline mozillians/jinja2/base.html]: Please fix identation.

**Comment by johngian** [comment]
Logged out page background is broken. (ignore the django error)
selection_024
selection_025

**Comment by comzeradd** [comment]
PR updated. Indentation was off all over the base template.

@johngian I'm having hard time to determine the problem at your last screenshots. Can you post a full page one.

**Comment by johngian** [comment]
selection_026
selection_027

**Comment by comzeradd** [comment]
PR updated with background color set back to sandstone for logged out home page.

**Comment by ppapadeas** [comment]
@comzeradd when logged out, navigating to a profile now displays wrong color for filler. See image
screen shot 2016-04-16 at 13 17 36'

Presumably this happened with the final fix for logged out main page. It should be filling with main color of the page, not hardcoded.

**Comment by comzeradd** [comment]
@ppapadeas good catch. PR updated. I fixed the `#page` background, since the loggedout homepage is the only place we use something else instead of white.

**Comment by akatsoulas** [comment]
@johngian could you please have another pass at this?

**Comment by johngian** [comment]
Please rebase

File: mozillians/jinja2/base.html [hunk 10]
```diff
@@ -57,127 +57,129 @@
...
-                          {% else %}
-                            {{ user.username }}
-                          {% endif %}
+            <ul id="nav-access">
+              <li><a href="#language-switcher">{{ _('Select language') }}</a></li>
+              <li><a href="#content">{{ _('Skip to main content') }}</a></li>
+            </ul>
+            {% block site_header_nav %}
+              <nav role="navigation">
+                  <a class="brand" href="{{ url('phonebook:home') }}">
```
Comment (johngian) [inline mozillians/jinja2/base.html]: Since you are fixing the indentation of this file please fix that one as well.

File: mozillians/jinja2/phonebook/home.html [hunk 11]
```diff
@@ -274,6 +254,30 @@
 
 {% endblock content %}
 
+{% block get_involved %}
```
Comment (johngian) [inline mozillians/jinja2/phonebook/home.html]: Since we are using this ``<article>`` only for non-authenticated user's home page, why are you moving it to it's own block that also adds the complexity to check if user is authenticated?

Comment (akatsoulas) [inline mozillians/jinja2/phonebook/home.html]: If this is not needed for the sticky footer, please revert this change.

Comment (comzeradd) [inline mozillians/jinja2/phonebook/home.html]: The "get involved" should be moved between `#page` and `footer`, thus the change. It can't live inside the content block.

**Comment by comzeradd** [comment]
PR rebased. I also fixed indentation issues on base template.

File: mozillians/jinja2/phonebook/home.html (hunk 11 above)
Comment (johngian) [inline mozillians/jinja2/phonebook/home.html]: sounds good.

**Comment by akatsoulas** [comment]
Tested this locally and looks good. r+ from me

**Comment by johngian** [comment]
On mobile view (360x640) ``#get_involved.content`` is misaligned to the left. It's on logged out home page.

**Comment by johngian** [comment]
r+wc from me

**Comment by comzeradd** [comment]
> On mobile view (360x640) #get_involved.content is misaligned to the left. It's on logged out home page.

I added an explicit `margin-left` to `#get_involved.content` to make sure it's rendered properly on mobile views.
